### PR TITLE
[Tool] turn on clang-tidy for all source code files

### PR DESCRIPTION
**Pull request by kevincai**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [x] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by dirtysalt — DISMISSED** [review]

File: gensrc/build/.clang-tidy [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (kevincai) [inline gensrc/build/.clang-tidy:1]: disable checks for generated source code.

File: be/src/formats/orc/apache-orc/.clang-tidy [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (kevincai) [inline be/src/formats/orc/apache-orc/.clang-tidy:1]: disable checks for apache-orc source code

File: be/src/gutil/strings/.clang-tidy [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (kevincai) [inline be/src/gutil/strings/.clang-tidy:1]: disable checks for gutil/strings

**Review by alvin-phoenix-ai — APPROVED** [review]

File: be/src/exec/avro_scanner.cpp [hunk 4]
```diff
@@ -144,7 +144,7 @@ Status AvroScanner::open() {
         }
 
         serdes_conf_t* sconf =
-                serdes_conf_new(NULL, 0, "schema.registry.url", confluent_schema_registry_url.c_str(), NULL);
+                serdes_conf_new(nullptr, 0, "schema.registry.url", confluent_schema_registry_url.c_str(), NULL);
```
Comment (trueeyu) [inline be/src/exec/avro_scanner.cpp:147]: NULL -> nullptr?

**Review by trueeyu — APPROVED** [review]

File: be/src/exec/avro_scanner.cpp (hunk 4 above)
Comment (kevincai) [reply be/src/exec/avro_scanner.cpp:147]: This is just the suggested fix by clang-tidy tool. not necessary to cover all the occurrence of NULL -> nullptr.

The goal is, after this PR, all the source code should be covered by the clang-tidy, and can be properly checked with new issues if any.

**Review by satanson — APPROVED** [review]

**Comment by kevincai** [comment]
@mergifyio backport branch-3.3

**Comment by kevincai** [comment]
give up backport to branch-3.3 for now.

**Pull request merged**